src/hooks/usePaginationFetch.ts

```diff
@@ -14,47 +17,57 @@ const usePaginationFetch = <T>(
   threshold: number = 0.5,
   errorMessage: string = "데이터를 불러오는데 실패했습니다.",
 ) => {
-  const [items, setItems] = useState<T[]>([]);
-  const [cursor, setCursor] = useState(0);
-  const [hasMoreList, setHasMoreList] = useState(true);
   const { ref: loader, isInView } = useInView<HTMLDivElement>(threshold);
 
-  const { fetchData, isLoading } = useFetch<PaginationData<T>>(url, {
+  const { fetchData } = useFetch<PaginationData<T>>(url, {
     autoFetch: false,
   });
 
-  const loadMoreItems = useCallback(async () => {
-    if (isLoading || !hasMoreList) return null;
+  const fetchPageData = useCallback(
+    async (context: QueryFunctionContext): Promise<PaginationData<T>> => {
+      const pageParam = (context.pageParam as number) ?? 0;
 
-    try {
-      const response = await fetchData(undefined, undefined, {
-        cursor,
+      const responseData = await fetchData(undefined, undefined, {
+        cursor: pageParam,
         limit,
       });
+      return responseData;
+    },
+    [limit, fetchData],
+  );
 
-      if (response.error) {
-        setHasMoreList(false);
-        return null;
-      }
-
-      if (response.data) {
-        setItems((prev) => [...prev, ...(response.data?.list ?? [])]);
-        setCursor(response.data.cursor);
-        setHasMoreList(response.data.hasMoreList);
-      }
-    } catch (error) {
-      console.error(errorMessage, error);
-      setHasMoreList(false);
+  const { data, isLoading, error, fetchNextPage, hasNextPage, isFetchingNextPage } = useInfiniteQuery<
+    PaginationData<T>,
+    ErrorData
+  >({
+    queryKey: ["themeProducts", url],
+    queryFn: fetchPageData,
+    getNextPageParam: (lastPage: PaginationData<T>) => {
+      return lastPage.hasMoreList ? lastPage.cursor : undefined;
+    },
+    initialPageParam: 0,
+  });
+
+  useEffect(() => {
+    if (error && isErrorData(error)) {
+      showFetchErrorToast(error.statusCode, errorMessage);
     }
-  }, [cursor, hasMoreList, isLoading, fetchData, limit, errorMessage]);
+  }, [error, errorMessage]);
 
   useEffect(() => {
-    if (isInView && hasMoreList && !isLoading) {
-      loadMoreItems();
+    if (isInView && hasNextPage && !isFetchingNextPage && !isLoading) {
```

Comment:
```suggestion
    if (isInView && hasNextPage && !isFetching) {
```

isFetchingNextPage, isLoading 이 둘을 isFetching 하나로 대신해도 되지 않을까요?